src/script/gen-corpus.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+# -*- mode:sh; tab-width:4; sh-basic-offset:4; indent-tabs-mode:nil -*-
+# vim: softtabstop=4 shiftwidth=4 expandtab
+
+set -ex
+
+function get_jobs() {
+    local jobs=$(nproc)
+    if [ $jobs -ge 8 ] ; then
+        echo 8
+    else
+        echo $jobs
+    fi
+}
+
+function build() {
+    local encode_dump_path=$1
+    shift
+
+    ./do_cmake.sh \
+        -DWITH_MGR_DASHBOARD_FRONTEND=OFF \
+        -DWITH_DPDK=OFF \
+        -DWITH_SPDK=OFF \
+        -DCMAKE_CXX_FLAGS="-DENCODE_DUMP_PATH=${encode_dump_path}"
+    cd build
+    cmake --build . -- -j$(get_jobs)
+}
+
+function run() {
+    MON=3 MGR=2 OSD=3 MDS=3 RGW=1 ../src/vstart.sh -n -x
+
+    local old_path="$PATH"
+    export PATH="bin:$PATH"
+    ceph osd pool create mypool 8
+    rados -p mypool bench 10 write -b 123
+    ceph osd out 0
+    ceph osd in 0
+    init-ceph restart osd.1
+    for f in ../qa/workunits/cls/*.sh ; do
+        $f
+    done
+    ../qa/workunits/rados/test.sh
+    ceph_test_librbd
+    ceph_test_libcephfs
```

Comment:
@cbodley do you know any simple way we can exercise rgw to generate some representative corpus objects?

Reply:
there's a qa/workunits/rgw/run-s3tests.sh that should give pretty good coverage - it may need some tweaks to pass, i'll give it a try. and it looks like you're already running ceph_test_cls_rgw in qa/workunits/cls/test_cls_rgw.sh :+1:

Reply:
i've made those tweaks to run-s3tests.sh in https://github.com/ceph/ceph/pull/28964. it `Ran 602 tests in 265.155s` - is that a reasonable amount of runtime for this script?

Reply:
thank you very much for enabling `run-s3tests.sh` to run with a vstart cluster, @cbodley !

yeah. considering that `import.sh` takes more than 30 minutes to finish, and we will run this script once every year on a single machine, it's completely acceptable!

i will add `run-s3tests.sh` to this script and update the document accordingly early tomorrow.